### PR TITLE
(chore) throttle deprecation messages

### LINKED ISSUE #3069: (chore) Throttle logging of deprecation messages
Would it maybe make sense to throttle logging the same deprecation messages? We're using highlight.js as a dependency of another dependency so can't directly update the deprecated code. The console is simply drowned in the exactly same deprecation message. Once would suffice, IMO, so we could alert the dependency to change it.

_Originally posted by @mihkeleidast in https://github.com/highlightjs/highlight.js/issues/2277#issuecomment-805046833_

### PR DESCRIPTION
**Pull request by mihkeleidast**
Resolves #3069.

### Changes

I've added a simple object map to keep track of already logged deprecations so they would not be logged multiple times.

I've manually tested that the following code now only logs one deprecation message:
```js
function highlight(code) {
  return hljs.highlight("java", code, false);
}

highlight(code);
highlight(code);
highlight(code);
```

### Checklist
- [x] Added markup tests, or they don't apply here because... does not change markup. could not find any existing logger tests so did not add any either.
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by mihkeleidast** [comment]
Looks like `depd` has limited or no browser support - what would be recommended to use here instead? [util-deprecate](https://www.npmjs.com/package/util-deprecate) looks popular but that does not have per-callsite logging.

File: src/highlight.js [hunk 1]
```diff
@@ -967,8 +964,7 @@ const HLJS = function(hljs) {
   @returns {string}
   */
   function deprecateFixMarkup(arg) {
-    logger.deprecated("10.2.0", "fixMarkup will be removed entirely in v11.0");
-    logger.deprecated("10.2.0", "Please see https://github.com/highlightjs/highlight.js/issues/2534");
+    logger.deprecated("10.2.0", "fixMarkup will be removed entirely in v11.0. Please see https://github.com/highlightjs/highlight.js/issues/2534");
```
Comment (joshgoebel) [inline src/highlight.js:967]: Please don't change the invokes, this is going to conflict with other work.  The only thing we should need to touch here if `logger.js`. Thanks.

Comment (mihkeleidast) [reply src/highlight.js]: Yeah no need for that if no need for depd.

**Review by joshgoebel — APPROVED** [review]

**Comment by joshgoebel** [comment]
@mihkeleidast Thanks so much!

**Pull request merged**